python-client/giskard/models/pytorch/__init__.py

```diff
@@ -0,0 +1,89 @@
+from typing import Union
+
+import mlflow
+import numpy as np
+import pandas as pd
+import torch
+from torch.utils.data import DataLoader
+from torch.utils.data import Dataset as torch_dataset
+
+from giskard.core.core import SupportedModelTypes
+from giskard.core.model import MLFlowBasedModel
+
+
+class TorchMinimalDataset(torch_dataset):
+    def __init__(self, df: pd.DataFrame, torch_dtype=torch.float32):
+        self.entries = df
+        self.torch_dtype = torch_dtype
+
+    def __len__(self):
+        return len(self.entries)
+
+    def __getitem__(self, idx):
+        return torch.tensor(self.entries.iloc[idx].to_numpy(), dtype=self.torch_dtype)
+
+
+class PyTorchModel(MLFlowBasedModel):
+
+    def __init__(self,
+                 clf,
+                 model_type: Union[SupportedModelTypes, str],
+                 torch_dtype=torch.float32,
```

Comment:
`torch_dtype` and `device` aren't saved and loaded back when you save/load the model